syntax = "proto3";

package frame.proto;

import "pixel.proto";

// Serialized cache for decoded image data.
message ImageCache {
    uint32 cache_version = 1;
    string source_relative = 2;
    uint64 source_size = 3;
    uint64 source_mtime_ns = 4;
    string cache_relative = 5;
    uint32 width = 6;
    uint32 height = 7;
    frame.proto.PixelElementSize pixel_element_size = 8;
    frame.proto.PixelStructure pixel_structure = 9;
    uint32 desired_channels = 10;
    bytes data = 11;
}
